### PR TITLE
Manage ColumnLayoutRelationData with a separate class

### PR DESCRIPTION
**Pull request by BUYT-1**

### CODE REVIEW

File: src/core/algorithms/fd/aidfd/aid.h [hunk 1]
```diff
@@ -38,6 +41,8 @@ class Aid : public FDAlgorithm {
 
     boost::dynamic_bitset<> constant_columns_;
 
+    void RegisterOptions();
```
[the remark this answers is not in the document]
Comment (BUYT-1) [reply src/core/algorithms/fd/aidfd/aid.h:44]: That's not possible, the base's method always gets called when in base's constructor.

[the remark this answers is not in the document]
Comment (Firsov62121) [reply src/core/algorithms/fd/aidfd/aid.h:44]: That's not a problem, make empty default implementation.

[the remark this answers is not in the document]
Comment (BUYT-1) [reply src/core/algorithms/fd/aidfd/aid.h:44]: [That implementation will be the one that's called from `FDAlgorithm`'s constructor, not the derived algorithm's one](https://isocpp.org/wiki/faq/strange-inheritance#calling-virtuals-from-ctors)

[the remark this answers is not in the document]
Comment (Firsov62121) [reply src/core/algorithms/fd/aidfd/aid.h:44]: Ok, my mistake.

**Review by Firsov62121 — APPROVED** [review]

**Review by Firsov62121 — APPROVED** [review]

**Review by Firsov62121 — APPROVED** [review]

**Review by chernishev — APPROVED** [review]
Approving instead of Mikhail Firsov

**Pull request merged**